app/models/user.rb

```diff
@@ -138,13 +143,47 @@ def active_for_authentication?
   def serving_transition_aged_youth?
     actively_assigned_and_active_cases.is_transitioned.any?
   end
+
+  def record_previous_email
+    if email_changed? && !old_emails.include?(email_was)
+      old_emails.push(email_was)
+    end
+  end
+
+  def skip_confirmable_email_confirmation_upon_creation
+    skip_confirmation!
+    confirm
+  end
+
+  def skip_casa_admin_email_changes
+    skip_reconfirmation!
+  end
+
+  def skip_email_changed_notification
+    true
+  end
+
+  def send_email_changed_notification?
+    # Don't send the notification if the skip flag is set
+    return false if skip_email_changed_notification
+
+    # Otherwise, let Devise send the notification
+    super
+  end
```

Comment:
I'm sure if I follow what's happening here.  `skip_email_changed_notificaton` is hard-coded to always return true, so the conditional will always read as: "return false if true"

From what I can tell, I don't think this would ever make it to the `super` call

Reply:
These methods were definitely not my best. I was struggling to get the proper user story for the confirmation flow: User submits email change -> confirmation email sent -> User confirms -> Old user email receives notification of email changed 

But I kept getting this type of flow:
User submits email change -> confirmation email sent & email changed notification sent to old email  -> User confirms (no email changed notification sent to old email after confirmation) 

I was confused because, after reading the Devise docs, I thought that final, post-confirmation email changed notification would automatically send but didn't. I ended up using ChatGPT to see if it could help and this method it generated seemed to work. I definitely don't have a solid understanding of ```super``` (this is a great excuse to jump into it more) and after doing some checks I trimmed what seemed to be the useless bits into only these two relevant methods:

```
def send_email_changed_notification #hardcoded in order to prevent the email changed notification BEFORE confirm
    return false 
  end

  def after_confirmation #sends the email changed notification AFTER user confirms
    send_email_changed_notification
  end
```

Having these two set up like this gets the user story I want but still seems kind of weird. I'm pretty sure they're related to ```Devise.config.send_email_changed_notification = true``` but would love to go over this with you.